syntax = "proto3";

service Crud {
    rpc CreateQueue (CreateRequest) returns (CreateReply);
    rpc ReadQueue (ReadRequest) returns (ReadReply);
    rpc DeleteQueue (DeleteRequest) returns (DeleteReply);
    rpc PutQueue (PutRequest) returns (PutReply);
    rpc GetQueues (GetRequest) returns (GetReply);
}

//Crud
message CreateRequest {
   string id = 1;
}

message CreateReply {
    string message = 1;
}

message ReadRequest {
   string id = 1;
}

message ReadReply {
    string message = 1;
}

message DeleteRequest {
   string id = 1;
}

message DeleteReply {
    string message = 1;
}

//Put/Get

message PutRequest {
   string id = 1;
   string content = 2;
}

message PutReply {
    string message = 1;
}

message GetRequest {}

message GetReply {
    string message = 1;
}